# Análisis Exploratorio de Datos (EDA)

## Autos usados en Lima – NeoAuto

Este cuaderno realiza un Análisis Exploratorio de Datos (EDA) sobre el dataset limpio `Datos_autos_limpios.csv`.

El objetivo del EDA es:

- Verificar la calidad y coherencia de los datos.

- Detectar valores atípicos o inconsistencias relevantes.

- Entender la estructura general del dataset antes de la visualización.

> 📌 Nota: Este EDA es intencionalmente ligero, ya que el análisis principal será presentado mediante un dashboard, que es el artefacto final orientado a stakeholders.


### 1. Carga de datos

In [8]:
import pandas as pd
import numpy as np

# Carga del dataset limpio
df = pd.read_csv("Datos_autos_limpios.csv")
df.head()


Marca_Modelo  Precio     Km              Transmision Combustible  \
0    Mitsubishi L200 2023   24000  11200                 Mecánica      Diesel   
1     Nissan Qashqai 2015   15500  83000               Automática    Gasolina   
2  Hyundai Grand I10 2016   10500  55000               Automática    Gasolina   
3         Bmw X3 30i 2024   52990  12800  Automática - Secuencial    Gasolina   
4      Ford Ecosport 2014   12900  87781                 Mecánica    Gasolina   

                                                 URL  Anio       Marca  \
0  https://neoauto.com/auto/usado/mitsubishi-l200...  2023  Mitsubishi   
1  https://neoauto.com/auto/usado/nissan-qashqai-...  2015      Nissan   
2  https://neoauto.com/auto/usado/hyundai-grand-i...  2016     Hyundai   
3  https://neoauto.com/auto/usado/bmw-x3-30i-2024...  2024         Bmw   
4  https://neoauto.com/auto/usado/ford-ecosport-2...  2014        Ford   

      Modelo  
0       L200  
1    Qashqai  
2  Grand I10  
3     X3 30i  
4   Ecosport

### 2. Visión general del dataset
Se revisa la dimensión del dataset y los tipos de datos para confirmar que el proceso de limpieza fue correcto.

In [9]:
df.shape
df.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 550 entries, 0 to 549
Data columns (total 9 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   Marca_Modelo  550 non-null    object
 1   Precio        550 non-null    int64 
 2   Km            550 non-null    int64 
 3   Transmision   550 non-null    object
 4   Combustible   550 non-null    object
 5   URL           550 non-null    object
 6   Anio          550 non-null    int64 
 7   Marca         550 non-null    object
 8   Modelo        550 non-null    object
dtypes: int64(3), object(6)
memory usage: 38.8+ KB


**Observaciones:**

- El dataset contiene 550 registros y 9 variables.

- No existen valores nulos.

- Las variables numéricas y categóricas están correctamente tipadas.

### 3. Estadísticas descriptivas 
Se analizan las principales métricas estadísticas de las variables numéricas.

In [10]:
df[["Precio", "Km", "Anio"]].describe()


Precio             Km         Anio
count     550.000000     550.000000   550.000000
mean    24489.461818   66444.009091  2017.732727
std     18801.128167   49723.778614     6.191202
min      3800.000000       0.000000  1967.000000
25%     14812.500000   34250.000000  2015.000000
50%     19500.000000   59000.000000  2018.000000
75%     28375.000000   86786.250000  2022.000000
max    198000.000000  350000.000000  2026.000000

**Observaciones clave:**

- El precio presenta una alta dispersión, coherente con la variedad de marcas y años.

- El kilometraje muestra valores elevados en algunos casos, esperable en autos usados.

- Se identifican valores atípicos en la variable Anio, que serán interpretados a nivel descriptivo y no corregidos en esta etapa.

### 4. Variables categóricas principales

**4.1 Marcas más frecuentes**

In [11]:
df["Marca"].value_counts().head(10)



Marca
Bmw           76
Toyota        38
Volkswagen    37
Audi          33
Mercedes      31
Mazda         29
Jeep          27
Subaru        26
Honda         25
Nissan        21
Name: count, dtype: int64

Este conteo permite identificar las marcas con mayor presencia en el mercado de autos usados en Lima.

**4.2 Tipo de transmisión**

In [12]:
df["Transmision"].value_counts()

Transmision
Automática - Secuencial    301
Automática                 145
Mecánica                   104
Name: count, dtype: int64

**4.3 Tipo de combustible**

In [13]:
df["Combustible"].value_counts()

Combustible
Gasolina     459
Diesel        35
Dual          35
Gas GLP        9
Gas GNV        6
Eléctrico      5
Gas            1
Name: count, dtype: int64

**Observaciones:**

- La gasolina es claramente el combustible predominante.

- Las categorías Gas GLP, Gas GNV y Gas aparecen con baja frecuencia.

- La categoría "Gas" con un solo registro probablemente corresponde a un error de clasificación del anunciante.

### 5. Análisis de correlación
Se analiza la relación lineal entre las variables numéricas.

In [14]:
df[["Precio", "Km", "Anio"]].corr()


Precio        Km      Anio
Precio  1.000000 -0.360574  0.310632
Km     -0.360574  1.000000 -0.530796
Anio    0.310632 -0.530796  1.000000

**Interpretación:**

- Existe una correlación negativa moderada entre Precio y Km, lo cual es consistente con el comportamiento del mercado.

- No se observan correlaciones fuertes entre el año del vehículo y el precio dentro de este dataset.

### 6. Conclusiones del EDA

- El dataset es consistente, limpio y usable para análisis.

- No presenta valores nulos ni errores críticos que impidan su uso.

- Las variables categóricas reflejan correctamente la estructura del mercado.

- Los datos están listos para su visualización y análisis exploratorio avanzado mediante dashboards.

📊 El análisis final y las respuestas a las preguntas de negocio se abordarán principalmente a través de visualizaciones interactivas.
